_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+    o[i.Name] = i;
+
+    return o;
+  }, {});
+
+  return layout[key].map(function(row) {
+    return row.map(function(image_name) {
+      return this.image_map[image_name];
+    }, {image_map: image_map});
+  }, {image_map: image_map});
+};
+
+const deepCopy = function(o) {
+  var output, v, key;
+  output = Array.isArray(o) ? [] : {};
+  for (key in o) {
+    v = o[key];
+    output[key] = (typeof v === 'object') ? deepCopy(v) : v;
+  }
+  return output;
+};
+
+const serialize = function(keys, state, delimit) {
+  return keys.reduce(function(h, k) {
+    var value = state[k] || 0;
+    // Array separated by underscore
+    if (value.constructor === Array) {
+      value = value.join('_');
+    }
+    return h + delimit + k + '=' + value;
+  }, '').slice(1);
+
+};
+
+const deserialize = function(entries) {
+  const query = entries.reduce(function(o, entry) {
+    if (entry) {
+      const kv = entry.split('=');
+      const val = kv.slice(1).join('=') || '1';
+      const vals = val.split('_');
+      const key = kv[0];
+      // Handle arrays or scalars
+      o[key] = vals.length > 1? vals: val;
+    }
+    return o;
+  }, {});
+
+  return query;
+};
+
+const newCopyYamlButton = function(THIS) {
+  const copy_pre = 'Copy to Clipboard';
+  const copy_post = 'Copied';
+  $(this).tooltip({
+    title: copy_pre
+  });
+
+  $(this).on('relabel', function(event, message) {
+    $(this).attr('data-original-title', message).tooltip('show');
+  });
+
+  $(this).click(function() {
+    $(this).trigger('relabel', [copy_post]);
+    ctrlC(THIS.bufferYaml);
+    setTimeout(function() {
+      $(this).trigger('relabel', [copy_pre]);
+    }, 1000);
+    return false;
+  });
+};
+
+const newCopyButton = function() {
+  const copy_pre = 'Copy to Clipboard';
+  const copy_post = 'Copied';
+  $(this).tooltip({
+    title: copy_pre
+  });
+
+  $(this).on('relabel', function(event, message) {
+    $(this).attr('data-original-title', message).tooltip('show');
+  });
+
+  $(this).on('click', function() {
+    const form = $(this).closest('form');
+    const formData = parseForm(form);
+    $(this).trigger('relabel', [copy_post]);
+    ctrlC(formData.copy_content);
+    setTimeout(function() {
+      $(this).trigger('relabel', [copy_pre]);
+    }, 1000);
+  return false;
+  });
+};
+
+const changeSprings = function(viewer, seconds, stiffness) {
+  const springs = [
+    'centerSpringX', 'centerSpringY', 'zoomSpring'
+  ];
+  springs.forEach(function(spring) {
+    const s = viewer.viewport[spring];
+    s.animationTime = seconds;
+    s.springStiffness = stiffness;
+    s.springTo(s.target.value);
+  });
+};
+
+const HashState = function(viewer, tileSources, exhibit, options) {
+
+  this.resetCount = 0;
+  this.embedded = options.embedded || false;
+  this.showdown = new showdown.Converter();
+  this.tileSources = tileSources;
+  this.exhibit = exhibit;
+  this.viewer = viewer;
+  viewer.setVisible(false);
+
+  this.hashable = {
+    exhibit: [
```

Comment:
The standard URL format has a story index "s", a waypoint index "w", a group index "g", and viewport coordinates "v".